src/box/lua/config/applier/console.lua

```diff
@@ -31,11 +33,19 @@ local function apply(config)
     local listen_uri = socket_file_to_listen_uri(socket_file)
     log.debug('console.apply: %s', listen_uri)
 
+    if old_socket ~= nil and old_socket:name() ~= socket_file then
+        if type(old_socket.close) == 'function' then
+            old_socket:close()
+            old_socket = nil
+        end
+    end
+
     -- Ignore 'Address already in use' error, because it is
     -- natural effect of reloading the configuration. All others
     -- errors are re-raised and lead to failure of apply of the
     -- configuration.
     local ok, res = pcall(console.listen, listen_uri)
+    old_socket = res
     if not ok and errno() ~= errno.EADDRINUSE then
         error(res)
     end
```

Comment:
Now we tracks the listening socket and there is no reason to perform `console.listen` if it already listens (and perform the `errno` check). We can just do the following (in pseudocode):

```lua
if socket_path_is_changed(<...>) then
    listening_socket:close()
    listening_socket = nil
end
if listening_socket == nil then
    listening_socket = console.listen(listen_uri)
end
```

Reply:
Please, leave comments open: I'll look over them at the next review iteration.

Reply:
Done